File: src/main/java/dev/slne/discord/spring/feign/dto/PunishmentBanDTO.java
package dev.slne.discord.spring.feign.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import javax.annotation.Nullable;
import lombok.AccessLevel;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;


/**
 * The type PunishmentBanDTO.
 */
@Getter
@ToString
@EqualsAndHashCode
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class PunishmentBanDTO {

  @JsonProperty("id")
  private long id;

  @JsonProperty("punishment_id")
  private String punishmentId;

  @JsonProperty("punished_uuid")
  private String punishedUuid;

  @Nullable
  @JsonProperty("server")
  private String server;

  @JsonProperty("security_ban")
  private boolean securityBan;

  @JsonProperty("permanent")
  private boolean permanent;

  @JsonProperty("raw_ban")
  private boolean rawBan;

  @Nullable
  @JsonProperty("reason")
  private String reason;

  @Nullable
  @JsonProperty("expiration_date")
  private String expirationDate;

  @Nullable
  @JsonProperty("issuer_uuid")
  private String issuerUuid;

  @JsonProperty("punishment_date")
  private String punishmentDate;

  @JsonProperty("unpunished")
  private boolean unpunished;

  @Nullable
  @JsonProperty("unpunished_date")
  private String unpunishedDate;

  @Nullable
  @JsonProperty("unpunished_issuer_uuid")
  private String unpunishedIssuerUuid;

  @JsonProperty("created_at")
  private String createdAt;

  @JsonProperty("updated_at")
  private String updatedAt;

}

File: src/main/kotlin/dev/slne/discord/spring/feign/dto/PunishmentBanDTO.kt
package dev.slne.discord.spring.feign.dto

import com.fasterxml.jackson.annotation.JsonProperty

/**
 * The type PunishmentBanDTO.
 */
@Getter
@ToString
@EqualsAndHashCode
@NoArgsConstructor(access = AccessLevel.PROTECTED)
class PunishmentBanDTO {
    @JsonProperty("id")
    private val id: Long = 0

    @JsonProperty("punishment_id")
    private val punishmentId: String? = null

    @JsonProperty("punished_uuid")
    private val punishedUuid: String? = null

    @Nullable
    @JsonProperty("server")
    private val server: String? = null

    @JsonProperty("security_ban")
    private val securityBan = false

    @JsonProperty("permanent")
    private val permanent = false

    @JsonProperty("raw_ban")
    private val rawBan = false

    @Nullable
    @JsonProperty("reason")
    private val reason: String? = null

    @Nullable
    @JsonProperty("expiration_date")
    private val expirationDate: String? = null

    @Nullable
    @JsonProperty("issuer_uuid")
    private val issuerUuid: String? = null

    @JsonProperty("punishment_date")
    private val punishmentDate: String? = null

    @JsonProperty("unpunished")
    private val unpunished = false

    @Nullable
    @JsonProperty("unpunished_date")
    private val unpunishedDate: String? = null

    @Nullable
    @JsonProperty("unpunished_issuer_uuid")
    private val unpunishedIssuerUuid: String? = null

    @JsonProperty("created_at")
    private val createdAt: String? = null

    @JsonProperty("updated_at")
    private val updatedAt: String? = null
}
